jerry-debugger/jerry_client_websocket.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import struct
+from jerry_client_tcp import Socket
+
+MAX_BUFFER_SIZE = 128
+WEBSOCKET_BINARY_FRAME = 2
+WEBSOCKET_FIN_BIT = 0x80
+
+class WebSocket(object):
+    def __init__(self, address, protocol=Socket()):
+
+        self.data_buffer = b""
+        self.protocol = protocol
+        self.address = address
+
+    def __handshake(self):
+        """ Client Handshake Request. """
+        self.__send_data(b"GET /jerry-debugger HTTP/1.1\r\n" +
+                         b"Upgrade: websocket\r\n" +
+                         b"Connection: Upgrade\r\n" +
+                         b"Sec-WebSocket-Key: dGhlIHNhbXBsZSBub25jZQ==\r\n\r\n")
+
+        # Expected answer from the handshake.
+        expected = (b"HTTP/1.1 101 Switching Protocols\r\n" +
+                    b"Upgrade: websocket\r\n" +
+                    b"Connection: Upgrade\r\n" +
+                    b"Sec-WebSocket-Accept: s3pPLMBiTxaQ9kYGzzhZRbK+xOo=\r\n\r\n")
+
+        len_expected = len(expected)
+
+        while len(self.data_buffer) < len_expected:
+            self.data_buffer += self.protocol.receive_data()
+
+        if self.data_buffer[0:len_expected] != expected:
+            raise Exception("Unexpected handshake")
```

Comment:
In the future it would be good to have specialized Exception classes so we can catch our module's exceptions in a uniform fashion and would make it possible to distinguish between the "base"  exceptions ans our exceptions.

Reply:
This applies to all exceptions in this PR.